Model/Schema.php

```diff
@@ -160,18 +160,37 @@ public function prepareOrderData(Order $order)
             $orderData['currency_iso'] = $order->getOrderCurrency()->getCode();
             $orderData['email'] = $order->getCustomerEmail();
             $orderData['customer_name'] = trim($order->getCustomerFirstName() . ' ' . $order->getCustomerLastName());
+            $orderData['platform'] = 'magento';
             if (!$orderData['customer_name'] && ($billingAddress = $order->getBillingAddress())) {
                 $orderData['customer_name'] = trim($billingAddress->getFirstname() . ' ' . $billingAddress->getLastname());
             }
             if (!$order->getCustomerIsGuest()) {
                 $orderData['user_reference'] = $order->getCustomerId();
             }
-            $orderData['platform'] = 'magento';
+            if (($fulfillmentDate = $this->getOrderFulfillmentDate($order))) {
+                $orderData['fulfillment_status'] = 'fulfilled';
+                $orderData['fulfillment_status_date'] = $fulfillmentDate;
+            }
         } catch (\Exception $e) {
             $this->yotpoConfig->log("Schema::prepareOrderData() - exception: " . $e->getMessage() . "\n" . $e->getTraceAsString(), "error");
             return [];
         }
 
         return $orderData;
     }
+
+    private function getOrderFulfillmentDate(Order $order)
+    {
+        $fulfillmentDate = null;
+        $lastStatus = $order->getStatusHistoryCollection()->getFirstItem();
+        foreach ($order->getStatusHistoryCollection() as $status) {
+            if ($status->getStatus() === Order::STATE_COMPLETE) {
+                return $status->getCreatedAt();
+            }
+            if (!$fulfillmentDate) {
```

Comment:
instead of (!$fulfillmentDate) you need to take the latest date so you need latest:
if (latsest < status->getCreatedAt())
 latest  =  status->getCreatedAt()

Reply:
It's already ordered by date, so the first one is the latest.
The script is checking if there's a "complete" status & if not, you'll get the latest change date.

Reply:
If latest date change can be "complete" as well, isn't enough to send the latest date change?

Reply:
There can be other status changes after the 'complete' depending on the merchant's setup. We can send the latest always, but as requested, we first look for a 'complete' & if it's missing we send the latest.

Reply:
I talked with our Product Manager and he prefer we will send the latest change date always, no dependency in the complete status.

Reply:
OK

Reply:
Changed it.